guides/authentication/mix_phx_gen_auth.md

```diff
@@ -0,0 +1,87 @@
+# mix phx.gen.auth
+
+The `mix phx.gen.auth` command generates a flexible, pre-built authentication system into your phoenix app. This simple generator allows you to quickly move past the task of adding authentication to your codebase and stay focused on the real-world problem your application is trying to solve.
+
+## Getting Started
+
+Let's start by running the following command from the root of our app (or `apps/my_app_web` in an umbrella app):
+
+    $ mix phx.gen.auth Accounts User users
+
+This creates an `Accounts` context with an `Accounts.User` schema module. The final argument is the plural version of the schema module which is used for generating database table names and route helpers. The `mix phx.gen.auth` generator is similar to `mix phx.gen.html` except it does not accept a list of additional fields to add to the schema and it generates many more context functions.
+
+Since this generator installed additional dependencies in `mix.exs`, let's fetch those dependencies:
+
+    $ mix deps.get
+
+Now we need to verify the database connection details for the development and test environments in `config/` so the migrator and tests can run properly. Then run the following to create the database:
+
+    $ mix ecto.setup
+
+Let's run the tests to make sure our new authentication system works as expected.
+
+    $ mix test
+
+And finally, let's start our phoenix server and try it out.
+
+    $ mix phx.server
+
+## Developer Responsibilities
+
+Since phoenix generates this code into your application instead of building these modules into phoenix itself, you now have complete freedom to modify the authentication system so it works best with your use case. The one caveat with using a generated authentication system is it will not be updated after it's been generated. Therefore as improvements are made to the output of `mix phx.gen.auth`, it becomes your responsibility to determine if these changes need to be ported into your application.
```

Comment:
```suggestion
Since phoenix generates this code into your application instead of building these modules into Phoenix itself, you now have complete freedom to modify the authentication system so it works best with your use case. The one caveat with using a generated authentication system is it will not be updated after it's been generated. Therefore as improvements are made to the output of `mix phx.gen.auth`, it becomes your responsibility to determine if these changes need to be ported into your application. Security-related and other important improvements will be explicitly and clearly marked in CHANGELOG and upgrade notes.
```